lamindb_setup/_init_instance.py

```diff
@@ -240,11 +241,7 @@ def infer_instance_name(
         # better way of accessing the database name?
         return str(db).split("/")[-1]
 
-    if isinstance(storage, str):
-        storage_path = StorageSettings._str_to_path(storage)
-    else:
-        storage_path = storage
-
+    storage_path = create_path(storage)
```

Comment:
I think `create_path` is not needed here, only `str_to_path` is enough.

Reply:
The way it's typed right now it also accepts UPath & Path. And str_to_path only accepts string, that's why I wrote it like this.

I agree that the additional checks aren't needed, but they also don't hurt because they're cached now.

I also agree that one could write this differently, I just wanted to eliminate the redundant if condition.

Reply:
ok, great, agree